packages/base-types/src/runtimeLogs/logs/steps/api.ts

```diff
@@ -0,0 +1,46 @@
+import { APIBodyType, APIMethod } from '@base-types/node/api';
+import { JsonArray, JsonObject, JsonValue } from 'type-fest';
+
+import { LoggingNodeType, LogLevel } from '../../utils';
+import { BaseStepLog } from '../base';
+
+interface ApiLogMessage {
+  request: {
+    method: APIMethod;
+    url: string;
+  };
+  response: {
+    statusCode: number;
+    statusMessage: string;
+  };
+}
+
+interface VerboseApiLogMessage {
+  request: {
+    headers: Record<string, string>;
+    query: Record<string, string>;
+  } & (
+    | {
+        method: Exclude<APIMethod, APIMethod.GET>;
+        bodyType: APIBodyType;
+        body: string | JsonObject | JsonArray;
```

Comment:
Might be simpler to always have this as a string. We can always parse it using the `bodyType`.

Reply:
i'm not sure if `bodyType` can be used for that. here's what the corresponding enum is defined as:
```ts
enum APIBodyType {
  FORM_DATA = 'formData',
  RAW_INPUT = 'rawInput',
  URL_ENCODED = 'urlEncoded',
}
```

although it is probably best to just make this a string of either JSON or whatever the `RAW_INPUT` may be (which could be JSON)

Reply:
`body` is now exclusively a `string`